cmd/kubectl-gadget/utils/trace.go

```diff
@@ -206,10 +206,6 @@ func createTraces(trace *gadgetv1alpha1.Trace) error {
 		}
 	}
 
-	// It is possible we change this value in the above loop, restore it to its
-	// previous value.
-	trace.Spec.Node = traceNode
```

Comment:
Are you sure this one is unnecessary?
Indeed, if you call this function with empty `trace.Spec.Node`, the loop will change its value each iteration.
So, I added to keep this value in the same state it was before the loop.

Reply:
I removed it because the caller is not using the trace after having deployed it (And I don't think it will). However, we can keep it anyway. I am OK either way.

Reply:
I tested this code, and it is indeed not needed, so good catch for spotting it!

Reply:
Is the caller always internal? If yes we can remove it.

Reply:
Yes, it is always internal, `createTraces` is not exposed as `utils` API.

Reply:
I confirm you this function (_i.e._ `createTraces`) is not exposed and only called by `CreateTrace()` (which is exposed).